### ISSUE TITLE
NJ Streamstatats

### ISSUE THREAD

**Issue by jknewson** [USGS-WiM/NSSServices#31]
I see that you have been working on fixing NJ in dev version 4, and that now it is producing low-flow estimates for both the base and the current periods. Thanks for your efforts on it. The site that you provided an output for in your last email had a drainage area that was lower than the minimum for the base period, so I selected another nearby site (40.27008, -74.46653) with a larger drainage area that was within the applicable range to see what the output would look like in that situation. There is still a problem with the output in that it is showing the error indicators as prediction errors, when in fact the values in SSDB are estimation errors, not prediction errors. Other than that, NJ now seems to be working correctly when only one hydrologic region is involved. 

It still is not working correctly when a basin is in both the coastal and non-coastal regions. See the attached example output. In the multi-region case, v4 is providing the flow estimates for each hydrologic region for both the base and current periods. However, it is providing only one set of area-averaged estimates, and those are not correct. In the attached example, the first statistic listed is the Jan 99 Pct Duration Min 1 Day Low Flow. The estimates for the two regions are correct, but v4 should be showing area-averaged estimates for both the current (27.0 cfs) and base periods (28.1 cfs). Instead, V4 is providing a single area-averaged estimate of 83.2 cfs. I have no clue how it is getting that number, which is much larger than it should be. Is there a way to fix this? If not, then we'll need to do one or a combination of the following: (1) shut off computation of the area-averaged estimates and provide a message saying that they need to be computed by hand, (2) continue to provide them but also provide a message saying that erroneous and they need to be computed by hand, or (3) set up exclusion areas that prevent computation of the low-flow stats when the basin for a site is in both hydrologic regions.

Also, in the attached output you will notice that the table of statistics for each month has the same citation listed four times. Is it possible to add a test to detect multiple instances of the same citation and show it only once? That would make the report substantially shorter.

[StreamStats 4_NJ_MultiRegionTest.pdf](https://github.com/USGS-WiM/NSSServices/files/739264/StreamStats.4_NJ_MultiRegionTest.pdf)

**Comment by USGSPMM3** [comment]
See email from pmccarth on 12/6/19. Also related to Iowa peak flow weighting issue and upcoming equations in Montana.

**Comment by HansVraga** [comment]
> I've been digging into old tickets and reviewing issues list in GitHub and have come across another issue with weighting in New Jersey that is similar to the Iowa weighting issue. I had previously thought that we should alter our database and services to look at the reference ID for weighting but the New Jersey example would still not result in correctly weighted values. Thus, I have two new ideas:
> IDEA 1 (temporary DB exercise)
> Create new flow stat labels in these special cases where multiple methods or equations compute statistics for essentially the same flow statistic. For example, in the case of New Jersey ( https://pubs.usgs.gov/sir/2014/5004/support/sir2014-5004.pdf ) there are equations for baseline conditions and equations for current conditions. I think we need to produce a set of baseline conditions statistic variables, and then alter the equations in NSS. The computations would then be properly weighted.
> 
> Similarly for Iowa, we have peak flow equations for all BCs and peak flow equations using DA only. I think we need to develop a flow stat variable for peak flow using DA only and then revise the equations accordingly. This would allow for proper weighting.
> 
> Soon to be coming in Montana we'll have peak flow equations based on bankfull parameters and these will need to have their own flow stat variable.  
> 
> IDEA 2 (Developer Exercise)
> Create an additional table in the database that groups region codes by type of statistic so the weighting procedure knows what regions codes can be weighted together. By default, nearly all region codes that share a common report reference and flow statistic variable can be populated together.  In the three cases above, we would manually "split" out the grouping in the database table. The weighting computations (is this in services or client) would then have to check the table for region codes/groups before weighting which is actually similar to it's current functionality (the way I understand it) but would be just a different table/check.
> 
> Idea 2 is a cleaner and better long-term solution, and a solution that we need to eventually implement. Given our backlog plan status I'm concerned that this solution could be pushed out beyond a year which makes me think that we need to initiate idea 1 until idea 2 can be implemented.
> 
> KJ, Jeremy, and Hans: Given our backlog is implementing idea 2 beyond a 1-year timeframe?

**Comment by kjacobsen16** [comment]
Rereading through this issue, making a note for myself to look at weighting with statistic groups. Bankfull statistics is an option, we could also make a "Drainage Area Only Statistics" group?
